### PR TITLE
[Merged by Bors] - refactor(Combinatorics/SimpleGraph): move `Walk` to its own file

### PR DESCRIPTION
**Pull request by Rida-Hamadani**
split `Path` and higher into a separate file from `Walk`

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Combinatorics/SimpleGraph/Connectivity.lean [hunk 1]
```diff
@@ -3,44 +3,13 @@ Copyright (c) 2021 Kyle Miller. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Kyle Miller
 -/
+import Mathlib.Combinatorics.SimpleGraph.Path
 import Mathlib.Combinatorics.SimpleGraph.Subgraph
 
 /-!
 
-# Graph connectivity
```
Comment (b-mehta) [inline Mathlib/Combinatorics/SimpleGraph/Connectivity.lean:10]: I think we can keep this title, but the rest of the deletions here are sensible

**Comment by YaelDillies** [comment]
Since `Walk` makes up the bulk of `SimpleGraph.Connectivity`, could you please first rename `SimpleGraph.Connectivity` to `SimpleGraph.Walk` in a separate PR? This should remove ~1k lines from the diff in this PR.

**Comment by Rida-Hamadani** [comment]
Sure! #15273

**Review by Parcly-Taxel — APPROVED** [review]

**Comment by kmill** [comment]
Thanks, this seems like a good direction. (And I would like to apologize to @Parcly-Taxel for dropping the ball with reviewing your splits from earlier this year — sorry!)

bors r+

**Pull request closed without merging**